### PR TITLE
Report problems applying changes

### PR DESCRIPTION
**Pull request by phil-davis**
At present various code keeps track of the return value of functions
that are called to apply changes ($retval). Then calls:
```get_std_save_message($retval)```
But get_std_save_message() always returns the same text "The changes
have been applied successfully."

So why is there are parameter to get_std_save_message()?

If we fix this up, returning different text to indicate that there is a
problem, then the subsequent print_info_box() that actually puts up the
message needs to pass the relevant severity also (success, warning,...).
So to minimize code that needs to be changed/duplicated in many places,
I have made print_apply_result_box() which takes $retval and sorts out
both the message text and severity all in 1 go.

system_routes has the typical changes that would be needed in the
top-level code to implement this.

Without changing top-level code, the changes in guiconfig.inc would
result in the problem text being reported when there is a problem, but
in the "success" color.

### CODE REVIEW

**Comment by phil-davis** [comment]
This is up for discussion, including:
1) The exact text of the message when there is a problem - is there always some evidence of the problem in System Logs somewhere?
2) Should the severity for the message box be "warning" or "danger" or?
3) Anything else?

File: src/usr/local/www/interfaces.php [hunk 1]
```diff
@@ -1695,11 +1698,10 @@ function build_gatewayv6_list() {
 					gettext("Don't forget to adjust the DHCP Server range if needed after applying."));
 }
 
-if ($savemsg) {
```
Comment (phil-davis) [inline src/usr/local/www/interfaces.php]: This savemsg code has been sitting unused, because the code for doing "Apply" never set up $savemsg.

**Comment by phil-davis** [comment]
The changes for interfaces.php are an example of a place where there is currently no "applied successfully" message displayed after pressing "Apply".
The changes in this example make the message happen and in the way that is integrated with the guiconfig.inc changes in this PR.

**Comment by phil-davis** [comment]
If the underlying infrastructure for this is agreed, I am happy to go through the GUI PHP files and make the small changes needed, like in the interfaces.php and system_routes.php examples.

**Comment by rbgarga** [comment]
I like the idea. What do you think @sbeaver-netgate and @jim-p ?

**Comment by phil-davis** [comment]
OK, I will look through the top-level UI code that uses get_std_save_message(), make the needed changes and push it...

**Comment by phil-davis** [comment]
Closing in favor of #3343 which has the (current) best/full solution.

**Pull request closed without merging**